tensorflow/c/logging.cc

```diff
@@ -0,0 +1,59 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include "tensorflow/c/logging.h"
+
+#include "tensorflow/core/platform/logging.h"
+#include "tensorflow/core/platform/stringprintf.h"
+
+static ::tensorflow::string BuildMessage(const char* fmt, va_list args) {
+  ::tensorflow::string message;
+  ::tensorflow::strings::Appendv(&message, fmt, args);
+  return message;
+}
+
+void TF_Log(TF_LogLevel level, const char* fmt, ...) {
+  if (level < 0 || level > 3) return;
```

Comment:
```suggestion
  if (level < TF_INFO || level > TF_FATAL) return;
```

Although, I would move this check to the `switch` `default` and `va_*` to `BuildMessage` and replace `<< message` with `<< BuildMessage(...` as this would be future proof in case logging levels get altered

Reply:
I am afraid that I do not understand what you means.
- I think we should just remove the test ?
- we have to keep va_* inside the `TF_Log` because there is no way to pass `...` to `BuildMessage`
Sorry if i have misunderstood something.

Reply:
I wanted something like

```cc
switch (level) {
  case TF_INFO:
    LOG(INFO) << BuildMessage(fmt, args);
    break;
  ...
  default: return;
  }
```

Without the initial `if`. But then `default: return` is also a no-op so I guess we can just remove the test?

But now I think it's better to just replace `0` and `3` with the values in the `enum`, like suggested and ignoring the rewrite suggested in the comment